core/events.md

```diff
@@ -0,0 +1,520 @@
+When they do appear they MUST use the JSONPath dot (`.`) notation to express
+the traversal to the attribute of interest.
+
+If a entity is updated for multiple reasons during the processing of an
+interaction, per the rules previously stated, only one `updated` event
+will be generated. This means the attribute names of all impacted attributes
+MUST be merged into one `changed` list.
+
+While `changed` is OPTIONAL, it is RECOMMENDED to be present due to its
+usefulness for consumers. However, if exposure of this information would be
+inappropriate for some scenarios then it MAY be excluded. For example, for
+privacy/security reasons.
+
+Implementations MAY define additional attributes as siblings to `changed`.
+
+This metadata (`data`) is NOT REQUIRED to be present in the event, but is
+RECOMMENDED.
+
+## Entity Events
+
+This section defines which `<ACTION>` values are applicable for each `<ENTITY>`
+value.
+
+### `registry` Events
+
+- Action: `updated`
+  - MUST be generated when a Registry's attribute is updated.
+  - MUST be generated when a Group is created or deleted, where `changed`
+    includes (at least) `epoch`, `modifiedat`, `<GROUPS>` and `<GROUPS>count`
+    attributes names.
+    TODO: GROUPS*??
+  - MUST be generated when `model`, `modelsource` or `capabilities` are
+    updated.
+
+The `created` or `updated` events are not defined as part of this specification
+as those operations are not defined in the xRegistry specification. It is
+expected that the tooling that performs those operations will define those
+events, if needed.
+
+### `model` Events
+
+- Action: `updated`
+  - TBD.
+
+### `modelsource` Events
+
+- Action: `updated`
+  - TBD.
+
+### `capabilities` Events
+
+- Action: `updated`
+  - TBD.
+
+### `group` Events
+
+- Action: `created`
+  - MUST be generated when a new Group is created.
+
+- Action: `updated`
+  - MUST be generated when a Group's attribute is updated.
+  - MUST be generated when a Resource is created or deleted, where `changed`
+    includes (at least) `epoch`, `modifiedat`, `<RESOURCES>` and
+    `<RESOURCES>count` attributes names.
+    TODO: RESOURCES*??
+
+- Action: `deleted`
+  - MUST be generated when a Group is deleted.
+
+### `resource` Events
+
+- Action: `created`
+  - MUST be generated when a new Resource is created.
+  - Note that at least one `io.xregistry.version.created` event will also be
+    being generated since at least one Version will also be created.
+
+- Action: `updated`
+  - MUST be generated when a Resource's (default Version) attribute is updated.
+  - MUST be generated when a Resource's `meta` attribute is updated, where
+    `changed` includes (at least) the changed top-level `meta` attributes
+    names prefixed with `meta.`. For example, `meta.defaultversionid`.
+    Note, this includes any updates to the `deprecated` sub-object, even
+    though a `deprecation` event is also generated.
+    TODO: meta.deprecated.* too??
```

Comment:
This depends on the resolution of the collection discussion, I'd keep it consistent with that decision.